src/app/auth/auth.js

```diff
@@ -0,0 +1,152 @@
+angular.module('prx.auth', ['prx.ui.nav'])
+.config(function (PRXDrawerProvider) {
+  PRXDrawerProvider.register({
+    name: 'Sign In',
+    weight: PRXDrawerProvider.BOTTOM,
+    href: 'http://www.prx.org/sessions/new',
+    nav: true,
+    template: '<prx-auth-badge></prx-auth-badge>',
+    dropdownTemplate: '<prx-auth-window></prx-auth-window>'
+  });
+}).service('PrxAuth', function (ngHal, $q, $window) {
+  var PrxAuth = this,
+      deferred = $q.defer(),
+      currentUser = {loggedIn: false};
```

Comment:
we always have a `currentUser` object because I want it to be a global singleton that I can update as the logged in state changes.

Rather than having `currentUser` be a falsey value when the user is not logged in, we have the `loggedIn` property of `currentUser` which is falsey. This allows us to change the `currentUser` object in a `$digest()` and keep the handles that any modules have on `currentUser` valid.